scripts/sync_tensorflow_protobuf_stubs.sh

```diff
@@ -1,73 +1,102 @@
 #!/bin/bash
-set -euxo pipefail
-
 # Partly based on scripts/generate_proto_stubs.sh.
-
 # Generates the protobuf stubs for the given tensorflow version using mypy-protobuf.
 # Generally, new minor versions are a good time to update the stubs.
-REPO_ROOT="$(realpath "$(dirname "${BASH_SOURCE[0]}")"/..)"
+set -euxo pipefail
 
-# This version should be consistent with the version in tensorflow's METADATA.toml.
-TENSORFLOW_VERSION=2.12.1
+# Need protoc >= 3.15 for explicit optional
+PROTOBUF_VERSION=25.3 # 4.25.3
+# Whenever you update TENSORFLOW_VERSION here, version should be updated
+# in stubs/tensorflow/METADATA.toml and vice-versa.
+TENSORFLOW_VERSION=2.16.1
 # Latest mypy-protobuf has dependency on protobuf >4, which is incompatible at runtime
```

Comment:
This might not be true anymore. Tensorflow has a pin to `<5.0.0dev`
https://github.com/tensorflow/tensorflow/blob/dab1b5bd71df9f2c1d1332e868b6c51422a00270/tensorflow/tools/pip_package/setup.py#L94

It's also completely mitigated by the venv anyway